spec/jobs/hyrax/grant_edit_job_spec.rb

```diff
@@ -1,11 +1,23 @@
 RSpec.describe Hyrax::GrantEditJob do
   let(:depositor) { create(:user) }
-  let(:file_set) { build(:file_set) }
 
-  it 'grants a user edit access to a FileSet' do
-    expect(FileSet).to receive(:find).with(file_set.id).and_return(file_set)
-    expect(file_set).to receive(:edit_users=).with(array_including(depositor.user_key))
-    expect(file_set).to receive(:save!)
-    described_class.perform_now(file_set.id, depositor.user_key)
+  context "when use_valkyrie is false" do
+    let(:file_set) { create(:file_set) }
+
+    it 'grants a user edit access to a FileSet' do
+      described_class.perform_now(file_set.id, depositor.user_key, use_valkyrie: false)
+      file_set.reload
+      expect(file_set.edit_users).to include depositor.user_key
+    end
+  end
+
+  context "when use_valkyrie is true" do
+    let(:file_set) { valkyrie_create(:hyrax_file_set) }
+
+    it 'grants a user edit access to a FileSet' do
+      described_class.perform_now(file_set.id, depositor.user_key, use_valkyrie: true)
```

Comment:
The id parameter should be a string to avoid serialization/deseriaization issues for that parameter when perform_later is used and the parameter goes on the message queue.